modules/renter/hostdb/scan.go

```diff
@@ -74,14 +70,33 @@ func (hdb *HostDB) queueScan(entry modules.HostDBEntry) {
 				entry = recentEntry
 			}
 
-			// Create new worker thread
+			// Check if a worker is ready to accept a new scanning job
+			select {
+			case scanPool <- entry:
+				hdb.mu.Unlock()
+				continue
+			default:
+			}
+
+			// Create new worker thread if work wasn't accepted immediately.
 			if hdb.scanningThreads < maxScanningThreads {
+				err := hdb.tg.Add()
+				if err != nil {
+					// Hostdb is shutting down
+					hdb.mu.Unlock()
+					return
+				}
+				// Spin up scanning thread
 				hdb.scanningThreads++
 				go func() {
 					hdb.threadedProbeHosts(scanPool)
 					hdb.mu.Lock()
 					hdb.scanningThreads--
+					// There is at least room for 1 new scanning thread now.
+					// Signal the world that a new thread can be started.
+					hdb.scanWait = false
```

Comment:
I think that this is incorrect. The purpose of the `scanWait` variable is to ensure that there's only one thread spinning up scanning threads. So the thread that should be setting `scanWait` to false is the thread that'sspinning up other threads, not the other threads being spun up. Here I think you end up settings `scanWait` to false a whole bunch of times, which can result in having many scanning pool threads.